SwSpotify/spotify.py

```diff
@@ -55,43 +61,22 @@ def get_info_mac():
 		end tell
 	end getCurrentlyPlayingTrack
 	"""
-	s = NSAppleScript.alloc().initWithSource_(apple_script_code)
-	x = s.executeAndReturnError_(None)
-	a = str(x[0]).split('"')
-	return a[1], a[3]
-
-
-def artist():
-	if platform.system() == "Windows":
-		try:
-			return get_info_windows()[0]
-		except:
-			return None
-	elif platform.system() == "Darwin":
-		try:
-			return get_info_mac()[0]
-		except:
-			return None	
-	else:
-		try:
-			return get_info_linux()[0]
-		except:
-			return None
-
-
-def song():
-	if platform.system() == "Windows":
-		try:
-			return get_info_windows()[1]
-		except:
-			return None
-	elif platform.system() == "Darwin":
-		try:
-			return get_info_mac()[1]
-		except:
-			return None	
-	else:
-		try:
-			return get_info_linux()[1]
-		except:
-			return None
+    s = NSAppleScript.alloc().initWithSource_(apple_script_code)
+    x = s.executeAndReturnError_(None)
+    a = str(x[0]).split('"')
+    return Song(a[1], a[3])
+
+
+def current_playing_song():
```

Comment:
what about something similar like 
```
def current():
    if platform.system() == "Windows":
        try:
            return get_info_windows()
        except:
            return None, None
    elif platform.system() == "Darwin":
        try:
            return get_info_mac()
        except:
            return None, None
    else:
        try:
            return get_info_linux()
        except:
            return None, None
```
so you could do tuple unpacking for the variables, as in:
`song, artist = current()`

Reply:
These are anti patterns
http://docs.quantifiedcode.com/python-anti-patterns/readability/not_using_named_tuples_when_returning_more_than_one_value.html
http://docs.quantifiedcode.com/python-anti-patterns/correctness/no_exception_type_specified.html

Reply:
> > Also, is using the python package better than using the default dbus?
> 
> dbus-python is now obsolete
> quote from official docs https://www.freedesktop.org/wiki/Software/DBusBindings/
> 
> > New applications should use pydbus, txdbus or GDBus/QtDBus bindings. See above.
> > dbus-python is a binding for libdbus, the reference implementation of D-Bus. For compatibility reasons, its API involves a lot of type-guessing (despite "explicit is better than implicit" and "resist the temptation to guess").

The idea behind using python3-dbus by default was because it's installed in every release by default. Is it the same as python-dbus?

Reply:
I think named tuples make sense where there's more of a need for readability while here you know that it's the song, artist that's going to be returned.

However, you have a fair point about adding the exception so go ahead with that!

Reply:
Named tuple inherit the tuple behavior https://dbader.org/blog/writing-clean-python-with-namedtuples so you also can use  argument unpacking with named tuple

Seems like information that dbus-python is obsolete was published between 2016 and 2018 and during this period there wasn’t releases https://dbus.freedesktop.org/doc/dbus-python/news.html